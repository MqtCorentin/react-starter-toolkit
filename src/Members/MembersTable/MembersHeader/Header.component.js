/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { Table } from '@axa-fr/react-toolkit-all';
import { NONE, DESCENDING, ASCENDING } from 'shared/constants';
import { TABLE_HEADERS_MEMBERS } from 'Members/constants';
import TableTh from './TableTh';
import { toggleSorting } from './Header.helper';

const TableHeader = ({ onSort, sorting, sortingInfo }) => (
  <Table.Header>
    <Table.Tr className="af-table__tr">
      {TABLE_HEADERS_MEMBERS.map(({ field, label }) => (
        <TableTh key={field} order={sortingInfo[field]} sort={() => onSort(toggleSorting(field, sorting))} sortable={Boolean(field)}>
          {label}
        </TableTh>
      ))}
    </Table.Tr>
  </Table.Header>
);

const propTypes = {
  onSort: PropTypes.func,
  sorting: PropTypes.object,
  sortingInfo: PropTypes.objectOf(PropTypes.oneOf([NONE, ASCENDING, DESCENDING])),
};

const defaultProps = {
  onSort: undefined,
  sorting: {},
  sortingInfo: {},
};

TableHeader.propTypes = propTypes;
TableHeader.defaultProps = defaultProps;

export default TableHeader;
