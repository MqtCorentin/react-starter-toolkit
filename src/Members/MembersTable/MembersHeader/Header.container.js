//
import React, { useContext } from 'react';
import { MembersContext } from 'Members/Members.container';
import TableHeader from './Header.component';

export const MembersHeaderEnhanced = () => {
  const { onChangeOrder, sorting } = useContext(MembersContext);
  const { field, order } = sorting;
  const sortingInfo = !field ? {} : { [field]: order };

  return <TableHeader sorting={sorting} sortingInfo={sortingInfo} onSort={onChangeOrder} />;
};

export default MembersHeaderEnhanced;
