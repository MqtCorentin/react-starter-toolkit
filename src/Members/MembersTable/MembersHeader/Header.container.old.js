import { withProps, compose } from 'recompose';
import withConsumerProps from 'Shared/hoc/withConsumerProps';
import withUpdate from 'Shared/hoc/withUpdate';
import TableHeader from './Header.component';
import { SimulationsListConsumer } from '../../SimulationsListContext';

export const withSortingConsumer = withConsumerProps(
  SimulationsListConsumer,
  ({ sorting, orderHandleOnChange }) => ({
    sorting,
    onSort: orderHandleOnChange,
  })
);

export const withSortingInfo = withProps(({ sorting: { field, order } }) => ({
  sortingInfo: !field ? {} : { [field]: order },
}));

const enhance = compose(withSortingConsumer, withSortingInfo, withUpdate);

export default enhance(TableHeader);
