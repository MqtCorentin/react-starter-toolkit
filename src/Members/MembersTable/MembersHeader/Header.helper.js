import { ASCENDING, DESCENDING } from 'shared/constants';

export const toggleOrder = order => (order === ASCENDING ? DESCENDING : ASCENDING);

export const toggleSorting = (field, { field: prevField, order }) => {
  const newOrder = {
    field,
    order: prevField !== field ? ASCENDING : toggleOrder(order),
  };
  return newOrder;
};
