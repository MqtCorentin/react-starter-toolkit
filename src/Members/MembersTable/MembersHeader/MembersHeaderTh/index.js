export { default } from './MembersHeaderTh.container';
