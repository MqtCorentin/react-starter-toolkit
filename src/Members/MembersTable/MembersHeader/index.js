export { default } from './MembersHeader.container';
