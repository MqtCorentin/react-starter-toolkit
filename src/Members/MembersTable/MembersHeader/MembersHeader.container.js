import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { MembersContext } from 'Members/Members.container';
import { TABLE_HEADERS_MEMBERS } from 'Members/constants';
import MembersHeader from './MembersHeader.component';

export const setSortingInfo = ({field, order}) => !field ? {} : { [field]: order };

export const MembersHeaderEnhanced = ({ setSortingInfoFn , headers}) => {
  const { onChangeOrder, sorting } = useContext(MembersContext);
  const { field, order } = sorting;
  const sortingInfo = setSortingInfoFn({field, order});

  return <MembersHeader sorting={sorting} sortingInfo={sortingInfo} onSort={onChangeOrder} headers={headers} />;
};

const MembersHeaderEnhancedPropTypes = {
  setSortingInfoFn: PropTypes.func,
  headers: PropTypes.arrayOf(
    PropTypes.shape({ 
      label: PropTypes.string.isRequired, 
      id: PropTypes.string.isRequired,
      field: PropTypes.string, 
    })
  )
};

const MembersHeaderEnhancedDefaultProps = {
  setSortingInfoFn: setSortingInfo,
  headers: TABLE_HEADERS_MEMBERS
};

MembersHeaderEnhanced.propTypes = MembersHeaderEnhancedPropTypes;
MembersHeaderEnhanced.defaultProps = MembersHeaderEnhancedDefaultProps;

export default MembersHeaderEnhanced;
