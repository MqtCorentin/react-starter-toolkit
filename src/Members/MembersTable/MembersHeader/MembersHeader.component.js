/* eslint-disable react/forbid-prop-types */
import React from 'react';
import PropTypes from 'prop-types';
import { Table } from '@axa-fr/react-toolkit-all';
import { NONE, DESCENDING, ASCENDING } from 'shared/constants';
import MembersHeaderTh from './MembersHeaderTh';

const MembersHeader = ({ onSort, sorting, sortingInfo, headers }) => (
  <Table.Header>
    <Table.Tr className="af-table__tr">
      {headers.map(({ field, label, id }) => (
        <MembersHeaderTh key={id} sorting={sorting} sortingInfo={sortingInfo} field={field} onSort={onSort}>
          {label}
        </MembersHeaderTh>
      ))}
    </Table.Tr>
  </Table.Header>
);

const propTypes = {
  onSort: PropTypes.func,
  sorting: PropTypes.object,
  sortingInfo: PropTypes.objectOf(PropTypes.oneOf([NONE, ASCENDING, DESCENDING])),
  headers: PropTypes.arrayOf(
    PropTypes.shape({
      label: PropTypes.string.isRequired,
      field: PropTypes.string,
      id: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

const defaultProps = {
  onSort: undefined,
  sorting: {},
  sortingInfo: {},
};

MembersHeader.propTypes = propTypes;
MembersHeader.defaultProps = defaultProps;

export default MembersHeader;
