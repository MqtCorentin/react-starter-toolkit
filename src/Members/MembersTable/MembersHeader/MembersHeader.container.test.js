import React from 'react';
import { render } from '@testing-library/react';
import { MembersHeaderEnhanced, setSortingInfo } from './MembersHeader.container';

describe('<MembersHeaderEnhanced />', () => {
  it('Should render MembersHeaderEnhanced', () => {
    const table = document.createElement('table');
    const defaultProps = { headers: [{ label: 'name', id: 'id'}]};
    const { asFragment } = render(<MembersHeaderEnhanced {...defaultProps} />, {
      container: document.body.appendChild(table),
    });
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('setSortingInfo', () => {
  it('Should return empty object when receive field is false', () => {
    const result = setSortingInfo({ order: 1 });
    expect(result).toEqual({});
  });

  it('Should return object with field key and order value when field is true', () => {
    const result = setSortingInfo({field: 'fieldname', order:1});
    expect(result).toEqual({fieldname: 1});
  });
});