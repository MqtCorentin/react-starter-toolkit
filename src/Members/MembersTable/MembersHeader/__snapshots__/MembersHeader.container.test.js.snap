// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MembersHeaderEnhanced /> Should render MembersHeaderEnhanced 1`] = `
<DocumentFragment>
  <thead
    class="af-table-thead"
  >
    <tr
      class="af-table__tr"
    >
      <th
        class="af-table__th"
        role="columnheader"
      >
        name
      </th>
    </tr>
  </thead>
</DocumentFragment>
`;
